rtgui/batchqueue.cc

```diff
@@ -99,20 +100,18 @@ BatchQueue::~BatchQueue ()
     }
 
     fd.clear ();
-
-    delete pmenu;
 }
 
 void BatchQueue::resizeLoadedQueue()
 {
-    // TODO: Check for Linux
 #if PROTECT_VECTORS
     MYWRITERLOCK(l, entryRW);
 #endif
 
-    for (size_t i = 0; i < fd.size(); i++) {
-        fd.at(i)->resize(getThumbnailHeight());
-    }
+    const auto height = getThumbnailHeight ();
+
+    for (const auto entry : fd)
```

Comment:
Yes, this is definitely one of the most useful small-scale additions. The syntax is almost equivalent, with a some caveats: `for(type var : container)` will basically expand to
````
for (auto iterator = std::begin(container), iterator != std::end(container); ++iterator) {
    type var = *iterator;
}
````
which should usually be more efficient than e.g. using indices. But it also means that one should usually bind `var` by reference, i.e. use `for(const auto& var : container)` as to not make a copy of `*iterator`. In this particular case, the entries are pointers so copying them is the same as taking a reference. (Also factoring out the common value of `getThumbnailHeight` is unrelated.)

Concerning the if clauses, this is the same as with the trailing underscores for member variables: I saw it in various places and hoped to be consistent by adopting it, but now I see that this is not the case. I will stick to braces for single line if clauses from now on, astyle will fix the rest.